### PR TITLE
Add `editor` folder

### PR DESCRIPTION
**Pull request by hunger**
... to collect documentation, configuration files, plugins, etc. for different
editors.

### CODE REVIEW

File: tools/lsp/README.md [hunk 1]
```diff
@@ -33,150 +33,7 @@ Alternatively, you can download one of our pre-built binaries for Linux or Windo
...
-   https://raw.githubusercontent.com/slint-ui/slint/master/tools/lsp/sublime/LSP.sublime-settings
-5. Modify the slint-lsp command path in `LSP.sublime-settings` to point to the cargo instalation path in your home folder (**Replace YOUR_USER by your username**):
-   `"command": ["/home/YOUR_USER/.cargo/bin/slint-lsp"]`
-6. Run "LSP: Enable Language Server Globally" or "LSP: Enable Lanuage Server in Project" from Sublime's Command Palette to allow the server to start.
-7. Open a .slint file - if the server starts its name will be in the left side of the status bar.
-
-In order to **preview a component**, when you have a .slint file open, place your cursor to
-the name of the component you would like to preview and select the "Show preview" button that
-will appear on the right of the editor pane.
+Please check the editor folder in the Slint repository for instructions on how to set up different editors to work with Slint.
```
Comment (ogoffart) [inline tools/lsp/README.md:39]: There could be a link here

**Review by ogoffart — COMMENTED** [review]
Good!

I wonder if we should from the `_extension` suffix from the vscode folder

**Comment by tronical** [comment]
I believe the paths in the vs code extension's `package.json` also need adjusting, in particular the `local-package` script

**Comment by tronical** [comment]
> I wonder if we should from the _extension suffix from the vscode folder

I like that idea!

File: editor/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+To show the preview, right click on the name definition of the component you want to preview
+(eg. `MainWindow` in `MainWindow := Window {`). Then in the menu, select *LSP Client > Code Action > Show Preview*.
+
+
+## QtCreator
+
+### Syntax Highlighting
+
+For the **syntax highlighting**, QtCreator supports the same format as Kate, with
+the [xml file](./slint.ksyntaxhighlighter.xml) at the same location.
```
Comment (ogoffart) [inline editor/README.md:56]: Link to be fixed

File: editor/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+Kate relies on the presence of syntax highlighting file for the usage of the LSP so we'll setup
+that first. The file [slint.ksyntaxhighlighter.xml](./slint.ksyntaxhighlighter.xml) needs to be copied
+in a location where kate can find it. See the [kate documentation](https://docs.kde.org/stable5/en/kate/katepart/highlight.html#katehighlight-xml-format)
+
+On Linux, this can be done by running this command
+
+```
+mkdir -p ~/.local/share/org.kde.syntax-highlighting/syntax/
+wget https://raw.githubusercontent.com/slint-ui/slint/master/tools/lsp/slint.ksyntaxhighlighter.xml -O ~/.local/share/org.kde.syntax-highlighting/syntax/slint.xml
```
Comment (ogoffart) [inline editor/README.md:21]: There too and a bunch of other occurrences

**Comment by ogoffart** [comment]
We also should look if we have links on our website or other locations that need to be changed

**Comment by hunger** [comment]
I caught another thing I needed to change in the vscode extionsion package.json file.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**